framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -785,6 +822,18 @@ protected void processCommonAssignment(TransferInput<V, S> in, Node lhs,
         // add new information based on conditional postcondition
         processConditionalPostconditions(n, method, tree, thenStore, elseStore);
 
+        if (inferSignatures) {
+            // Finds the receiver's type
+            Tree receiverTree = n.getTarget().getReceiver().getTree();
+            if (receiverTree == null) {
+                // If there is no receiver, then get the class being visited.
+                receiverTree = analysis.getContainingClass(n.getTree());
```

Comment:
Is this in order to cover the case of a method called in a static context? If so, do you still need the "TODO: Method called from static context." in SignatureInferenceScenes.updateInferredMethodParametersTypes?